libraries/net/message_oriented_connection.cpp

```diff
@@ -63,7 +63,7 @@ namespace graphene { namespace net {
       fc::time_point _last_message_sent_time;
 
       bool _send_message_in_progress;
```

Comment:
While you're on it, perhaps update `_send_message_in_progress` to atomic and update the `assert` in the guarding code to `FC_ASSERT`?

Reply:
Replace `verify_no_send_in_progress` with `no_parallel_execution_guard`?

Reply:
Yes.

Reply:
done